light_painting/scripts/serial_node.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+
+#####################################################################
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2011, Willow Garage, Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of Willow Garage, Inc. nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+
+import rospy
+from rosserial_python import SerialClient, RosSerialServer
+from serial import SerialException
+from time import sleep
+import multiprocessing
+import serial
+import serial.tools.list_ports
+import sys
+
+
+if __name__ == "__main__":
+   rospy.init_node("serial_node")
+   UNO_VENDOR_ID = 0X2341
```

Comment:
Thinking aloud on robustness of this technique. From what I now understand via some quick searches, the Vendor ID is owned by Arduino ([reference](https://learn.adafruit.com/arduino-tips-tricks-and-techniques/arduino-uno-faq#faq-1884385)).

It also seems a second ID exists, but hopefully we don't run into that..

```
arduino.cc : 9025(0x2341)
arduino.org : 10755(0x2A03)
```

From this database, I found the latest Arduino device ID's broken out. http://www.linux-usb.org/usb-ids.html
```
2341  Arduino SA
	0001  Uno (CDC ACM)
	0010  Mega 2560 (CDC ACM)
	0036  Leonardo Bootloader
	003b  Serial Adapter (CDC ACM)
	003d  Due Programming Port
	003e  Due
	003f  Mega ADK (CDC ACM)
	0042  Mega 2560 R3 (CDC ACM)
	0043  Uno R3 (CDC ACM)
	0044  Mega ADK R3 (CDC ACM)
	0045  Serial R3 (CDC ACM)
	0049  ISP
	8036  Leonardo (CDC ACM, HID)
	8038  Robot Control Board (CDC ACM, HID)
	8039  Robot Motor Board (CDC ACM, HID)
```

Based on this, I think there could be some breaking behavior if an older (rev 1 or rev 2) Arduino Uno is used. For our internal lab purposes, I don't believe this is an issue. But this should be documented.